src/main/java/seedu/address/logic/commands/client/DeleteClientCommand.java

```diff
@@ -34,17 +40,69 @@ public DeleteClientCommand(Index targetIndex) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Client> lastShownList = model.getFilteredClientList();
+        List<Client> lastShownClientList = model.getFilteredClientList();
+        List<Deal> lastShownDealList = model.getFilteredDealList();
+        List<Event> lastShownEventList = model.getFilteredEventList();
+        List<Property> lastShownPropertyList = model.getFilteredPropertyList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+        if (targetIndex.getZeroBased() >= lastShownClientList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_CLIENT_DISPLAYED_INDEX);
         }
 
-        Client clientToDelete = lastShownList.get(targetIndex.getZeroBased());
+        Client clientToDelete = lastShownClientList.get(targetIndex.getZeroBased());
+
+        if (existInDeal(clientToDelete, lastShownDealList)
+            || existInEvent(clientToDelete, lastShownEventList)
+            || existInProperty(clientToDelete, lastShownPropertyList)) {
+            throw new CommandException("Cannot delete");
+        }
+
         model.deleteClient(clientToDelete);
         return new CommandResult(String.format(MESSAGE_DELETE_CLIENT_SUCCESS, Messages.formatClient(clientToDelete)));
     }
 
+    private boolean existInDeal(Client clientToDelete, List<Deal> dealList) {
+        boolean isInDeal = false;
+        ClientName clientNameToDelete = clientToDelete.getClientName();
+        for (Deal deal : dealList) {
+            if (clientNameToDelete.equals(deal.getBuyer()) || clientNameToDelete.equals(deal.getSeller())) {
+                if (deal.getStatus() == DealStatus.CLOSED) {
+                    continue;
+                }
+                isInDeal = true;
+                break;
+            }
+        }
+        return isInDeal;
+    }
```

Comment:
Why not just use the boolean values directly?
```suggestion
    private boolean existInDeal(Client clientToDelete, List<Deal> dealList) {
        ClientName clientNameToDelete = clientToDelete.getClientName();
        for (Deal deal : dealList) {
            if ((clientNameToDelete.equals(deal.getBuyer()) || clientNameToDelete.equals(deal.getSeller()))
                && deal.getStatus() != DealStatus.CLOSED) {
                    return true;
            }
        }
        return false;
    }
```